File: 小码哥大神5期练习/5、Runtime&RAC&MVVM&新特性/1025/函数编程思想的计算器/函数编程思想的计算器/ViewController.m
//
//  ViewController.m
//  函数编程思想的计算器
//
//  Created by coco on 16/7/18.
//  Copyright © 2016年 XHJ. All rights reserved.
//

#import "ViewController.h"
#import "Calculation.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    Calculation *calculation = [[Calculation alloc] init];
    float result = [calculation Calculation:^float(float result) {
        result += 10;
        result -= 4;
        result *= 2.1;
        result /= 3;
        return result;
    }].result;
    NSLog(@"%f", result);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 小码哥大神5期练习/自定义Log/自定义Log/ViewController.swift
//
//  ViewController.swift
//  自定义Log
//
//  Created by M-coppco on 16/8/9.
//  Copyright © 2016年 M-coppco. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.

        
        if let temp = NSBundle.mainBundle().infoDictionary {
            let name = temp["CFBundleExecutable"]
            print(name)
        }
        
        print(product_name!)
        let cls = NSClassFromString("自定义Log." + "ViewController")
        let clsType = cls as! UIViewController.Type
        
        HJLog(clsType.init())
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

